pimd/pim_upstream.c

```diff
@@ -207,9 +215,7 @@ struct pim_upstream *pim_upstream_del(struct pim_instance *pim,
 	if (pim_up_mlag_is_local(up))
 		pim_mlag_up_local_del(pim, up);
 
-	THREAD_OFF(up->t_ka_timer);
-	THREAD_OFF(up->t_rs_timer);
-	THREAD_OFF(up->t_msdp_reg_timer);
+	pim_upstream_timers_stop_helper(up);
```

Comment:
`pim_upstream_timers_stop_helper` should be put before checking ref counter imo.

Reply:
that is a really bad idea in that it will absolutely kill pim's ability to do any work when a if channel is deleted( as an example ).

Reply:
Yes, we should keep KAT running.
We all agree that KAT should not stopped until  instance terminate.
But since “no ip pim” will not enter “pim_upstream_terminate”, it can’t help my crash issue. Maybe my PR#10569 is ok.